tests/terraform/README.md

```diff
@@ -28,9 +39,54 @@ $ docker build . -f ./tests/terraform/scripts/Dockerfile.build -t rke2-tf
 $ docker run --name rke2-tf-creation-test -t -e AWS_ACCESS_KEY_ID=<YOUR_ACCESS_KEY> -e AWS_SECRET_ACCESS_KEY=<YOUR_SECRET_KEY> -v /path/to/aws/key.pem:/tmp/aws_key.pem rke2-tf sh -c "go test -timeout=30m -v ./tests/terraform/createcluster/..."
 $ docker run --name rke2-tf-upgrade-test -t -e AWS_ACCESS_KEY_ID=<YOUR_ACCESS_KEY> -e AWS_SECRET_ACCESS_KEY=<YOUR_SECRET_KEY> -v /path/to/aws/key.pem:/tmp/aws_key.pem rke2-tf sh -c "go test -timeout=45m -v ./tests/terraform/upgradecluster/... -upgradeVersion=v1.24.8+rke2r1"
 ```
+Test Flags:
+```
+- ${upgradeVersion} version to upgrade to
+```
+We can also run tests through the Makefile through tests' directory:
+```bash
+Args:
+*All args are optional and can be used with:
+
+`$make tf-tests-run`   `$make tf-tests-logs`,
+`$make vet-lint`       `$make tf-tests-complete`, 
+`$make tf-upgrade`     `$make tf-test-suite-same-cluster`,
+`$make tf-test-suite`
+
+- ${IMGNAME}     append any string to the end of image name
+- ${ARGNAME}     name of the arg to pass to the test
+- ${ARGVALUE}    value of the arg to pass to the test
+- ${TESTDIR}     path to the test directory 
+
+Commands:
+$ make tdf-tests-up                  # create the image from Dockerfile.build
+$ make tf-tests-run                  # runs all tests if no flags or args provided
+$ make tf-tests-down                 # removes the image
+$ make tf-tests-clean                # removes instances and resources created by tests
+$ make tf-tests-logs                 # prints logs from container the tests
+$ make tf-tests-complete             # clean resources + remove images + run tests
+$ make vet-lint                      # runs go vet and go lint
+$ make tf-create                     # runs create cluster test locally
+$ make tf-upgrade                    # runs upgrade cluster test locally
+$ make tf-test-suite-same-cluster    # runs all tests locally in sequence using the same state    
+$ make tf-remove-state               # removes terraform state dir and files
+$ make tf-test-suite                 # runs all tests locally in sequence not using the same state
+      
+Examples:
+$ make tf-tests-run IMGNAME=2 TESTDIR=terraform/upgradecluster ARGNAME=upgradeVersion ARGVALUE=v1.26.2+rke2r1
+$ make tf-tests-run TESTDIR=terraform/upgradecluster
+$ make tf-tests-logs IMGNAME=1
+$ make vet-lint TESTDIR=terraform/upgradecluster
+```
+
+
+In between tests, if the cluster is not destroyed, then make sure to delete the ./tests/terraform/modules/terraform.tfstate + .terraform.lock.hcl file if you want to create a new cluster.
 
-In between tests, if the cluster is not destroyed, then make sure to delete the ./tests/terraform/modules/terraform.tfstate file if you want to create a new cluster.
 
+# Common Issues:
+````
+- Issues related to terraform plugin please also delete the /.terraform folder and or go to folder that have main.tf and just run `terraform init` to download the plugin again
```

Comment:
Did you find that you had to run `terraform init` locally at all? That should be happening during the execution via go automatically from the terraform package.

Reply:
nope, its not running the apply myself for the run .... sometimes on my mac m1 terraform returns some versions errors that can only be fixed doing this init on the main, but whitout any configuration, just run a init to download newer thing.

but i am not being able to reproduce again so i will take it off from the README and if needed we can add it later